src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -247,6 +247,83 @@
  *         }
  *     }
  *
+ * ## Tokens and finer-grained processing.
+ *
+ * It's also possible to scan through every lexical token in
+ * the HTML document using the `next_token()` function. This
+ * alternative form takes no argument and provides no built-in
+ * query syntax.
+ *
+ * Example:
+ *
+ *      $title        = '(untitled)';
+ *      $text_content = '';
+ *      while ( $processor->next_token() ) {
+ *          switch ( $processor->get_token_name() ) {
+ *              case '#text':
+ *                  $text .= $processor->get_node_text();
+ *                  break;
+ *
+ *              case 'BR':
+ *                  $text .= "\n";
+ *                  break;
+ *
+ *              case 'TITLE':
+ *                  $title = $processor->get_node_text();
+ *                  break;
+ *          }
+ *      }
+ *      return trim( "# {$title}\n\n{$text_content}\n" );
```

Comment:
Just flagging that there are two different variables for the text, `$text` and `$text_content`. Also, acknowledging that it's only an example, it reads a little strangely to append `\n` only to then run `trim()`.

Reply:
good catch @dlh01 - I updated this example in 8793d47fdec323539ccfc58efb4103fb7a864a6d